.github/workflows/reusable-build-test-release.yml

```
@@ -45,8 +45,93 @@ on:
         required: false
 
 jobs:
+  setup-workflow:
+    runs-on: ubuntu-latest
+    outputs:
+      skip-workflow: ${{ steps.skip-workflow.outputs.skip-workflow }}
+      delay-destroy-ko: ${{ steps.delay-destroy-setup.outputs.delay-destroy-ko }}
+      delay-destroy-ui: ${{ steps.delay-destroy-setup.outputs.delay-destroy-ui }}
+      delay-destroy-modinput_functional: ${{ steps.delay-destroy-setup.outputs.delay-destroy-modinput_functional }}
+      delay-destroy-escu: ${{ steps.delay-destroy-setup.outputs.delay-destroy-escu }}
+      delay-destroy-scripted_inputs: ${{ steps.delay-destroy-setup.outputs.delay-destroy-scripted_inputs }}
+      delay-destroy-requirement_test: ${{ steps.delay-destroy-setup.outputs.delay-destroy-requirement_test }}
+      execute-ko: ${{ steps.delay-destroy-setup.outputs.execute-ko }}
+      execute-ui: ${{ steps.delay-destroy-setup.outputs.execute-ui }}
+      execute-escu: ${{ steps.delay-destroy-setup.outputs.execute-escu }}
+      execute-modinput_functional: ${{ steps.delay-destroy-setup.outputs.execute-modinput_functional }}
+      execute-scripted_inputs: ${{ steps.delay-destroy-setup.outputs.execute-scripted_inputs }}
+      execute-requirement_test: ${{ steps.delay-destroy-setup.outputs.execute-requirement_test }}
+    steps:
+      - name: skip workflow if description is empty for labeled pr
+        id: skip-workflow
+        env:
+          PR_BODY: ${{ github.event.pull_request.body }}
+        run: |
+          set +e
+          TESTSET="knowledge ui modinput_functional scripted_inputs escu requirement_test"
+          echo "testset=$TESTSET" >> "$GITHUB_OUTPUT"
+          SKIP_WORKFLOW="No"
+          if [[ '${{ github.event.action }}' == 'labeled' && '${{ github.event.label.name }}' == 'preserve_infra' ]]; then
+            echo "$PR_BODY" >> body.txt
+            SKIP_WORKFLOW="Yes" 
+            tests=$(grep -i "^preserve:" body.txt | { grep -v grep || true; })
+            for test_type in $TESTSET; do
+                if [[ $tests =~ $test_type ]]; then
+                    eval SKIP_WORKFLOW="No"
+                fi
+            done
+          fi
+          echo "skip-workflow=$SKIP_WORKFLOW" >> "$GITHUB_OUTPUT"
+          if [ "$SKIP_WORKFLOW" == "Yes" ]; then 
+            echo "No description is provided with preserve infra label"
+          fi 
+      - name: setup for delay destroy
+        id: delay-destroy-setup
+        shell: bash
+        env:
+          PR_BODY: ${{ github.event.pull_request.body }}
+        run: |
+          set +e
+          TESTSET="${{ steps.skip-workflow.outputs.testset }}"
+          for i in $TESTSET; do
+              eval DELAY_DESTROY_"$i"="No"
+              eval EXECUTE_"$i"="Yes"
```

It should also work I guess
```suggestion
              DELAY_DESTROY_"$i"="No"
              EXECUTE_"$i"="Yes"
```

---

Without eval, this command will throw error like `DELAY_DESTROY_knowledge=No: command not found` since we are using a variable DELAY_DESTORY_ and EXECUTE_ while assigning the value.